### PR TITLE
Feat: Rhai support

### PR DESCRIPTION
**Pull request by thedanvail**
This is very much a draft PR and will need a lot of back-and-forth as I work to understand the intricacies of the project, but I wanted to check this in so @makspll can comment/let me know thoughts as this progresses.

### CODE REVIEW

File: languages/bevy_mod_scripting_rhai_derive/src/rhai_method.rs [hunk 1]
```diff
@@ -1,29 +1,113 @@
...
+            is_function: false,
+            field_function: RhaiFieldFunction::Setter,
+            method_type: RhaiMethodMetaType::Static,
+            method_name: String::from("yeet"),
             method: input.parse()?,
         })
     }
 }
 
+pub(crate) struct RhaiClosure {
```
Comment (makspll) [inline languages/bevy_mod_scripting_rhai_derive/src/rhai_method.rs:74]: So yes in the current macro implementation I create new syn types, however, I believe those really polluted the code and made things more convoluted. In the new Lua implementation I am simply using existing syn types such as `TraitItemMethod` and simply using the available attribute slots to add metadata for each language, which seems to work better with IDE's as well

File: languages/bevy_mod_scripting_rhai_derive/src/rhai_method.rs [hunk 2]
```diff
@@ -1,29 +1,113 @@
...
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let arrow = &self.arrow;
+        let expr = &self.expr;
+        tokens.extend(quote::quote! {
+            #arrow #expr
+        });
+    }
+}
+
+pub(crate) struct RhaiMethod {
```
Comment (makspll) [inline languages/bevy_mod_scripting_rhai_derive/src/rhai_method.rs:98]: Similar to above

**Review by makspll — COMMENTED** [review]
Sorry for delay, here are some comments, I've pushed some commits to the `feature/derive_macros` branch as well for the Lua side of things which hopefully illustrate these comments :)

**Comment by thedanvail** [comment]
> Sorry for delay, here are some comments, I've pushed some commits to the `feature/derive_macros` branch as well for the Lua side of things which hopefully illustrate these comments :)

Excellent. Thanks. For the sake of staying in line with the main repo, would you mind granting me write permissions so I can develop on a branch such as `feature/rhai_support`?

**Pull request closed without merging**